syntax = "proto3";

package agendamento;

message Consulta {
  string id_consulta = 1; // O código único
  string paciente = 2;
  string medico = 3;
  string data = 4;
  string horario = 5;
}

message AgendarConsultaRequest {
  // Não precisa do ID aqui, o servidor vai gerar
  string paciente = 1;
  string medico = 2;
  string data = 3;
  string horario = 4;
}

message AgendarConsultaResponse {
  bool sucesso = 1;
  string mensagem = 2;
  string id_consulta_gerado = 3; // Devolve o código para o usuário
}

message GerenciarConsultaRequest {
  string id_consulta = 1;
}

message GerenciarConsultaResponse {
  bool sucesso = 1;
  string mensagem = 2;
  Consulta consulta = 3; // Devolve a consulta se for uma busca
}

service AgendamentoMedico {
  // Cadastra uma nova consulta e retorna um código único
  rpc AgendarConsulta(AgendarConsultaRequest) returns (AgendarConsultaResponse);

  // Busca os detalhes de uma consulta usando seu código
  rpc BuscarConsulta(GerenciarConsultaRequest) returns (GerenciarConsultaResponse);

  // Cancela uma consulta usando seu código
  rpc CancelarConsulta(GerenciarConsultaRequest) returns (GerenciarConsultaResponse);
}